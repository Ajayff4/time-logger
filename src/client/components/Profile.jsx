import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import { profileLoad } from '../actions'
import PropTypes from 'prop-types'

const Profile = (props) => {
    useEffect(() => {
        props.profileLoad()
    })
    return (
        <div>
            <h2>Profile Component</h2>
            <h3>Username: {props.username}</h3>
            <h3>Entries made: {props.logs ? props.logs.length : 0}</h3>
        </div>
    )
}

Profile.propTypes = {
    username: PropTypes.string.isRequired,
    logsCount: PropTypes.number.isRequired,
    profileLoad: PropTypes.func.isRequired
}

const mapStateToProps = state => {
    return {
        username: state.user.username,
        logs: state.log.logs
    }
}

const mapDispatchToProps = dispatch => {
    return {
        profileLoad: () => dispatch(profileLoad())
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Profile)