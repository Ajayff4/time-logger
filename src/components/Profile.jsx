import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class Profile extends Component {
    static propTypes = {
        username: PropTypes.string,
        fullname: PropTypes.string,
        email: PropTypes.string,
        logs: PropTypes.array
    }
    render() {
        let completed = 0
        let pending = 0
        let dayLogTime = 0
        let totalLogTime = 0
        if(this.props.logs!==undefined && this.props.logs.length>0){
            //eslint-disable-next-line
            this.props.logs.map(log => {
                //eslint-disable-next-line
                log.completed==1? completed++ : pending++;
                
                let today = new Date().toLocaleDateString().split("/")
                today = today[2] + "-" + today[1] + "-" + today[0];
                if(log.date===today.toString()) {
                    dayLogTime += log.duration
                }
                totalLogTime += log.duration
            })
        }
        //const tableComponent = ;
        return (
            <div id="profileComponent">
                <center>
                    <div className="container">
                        <table className="table table-dark table-striped">
                        <tbody>
                            <tr key="header">
                                <th>#</th><th>Info</th><th>Description</th>
                            </tr>
                            <tr key="username">
                                <td>1</td><td>Username</td><td>{this.props.username}</td>
                            </tr>
                            <tr key="fullname"> 
                                <td>2</td><td>Fullname</td><td>{this.props.fullname}</td>
                            </tr>
                            <tr key="email">
                                <td>3</td><td>Email</td><td>{this.props.email}</td>
                            </tr>
                            <tr key="completed">
                                <td>4</td><td>Completed</td><td>{completed>1 ? `${completed} logs`:`${completed} log`}</td>
                            </tr>
                            <tr key="pending">
                                <td>5</td><td>Pending</td><td>{pending>1 ? `${pending} logs`:`${pending} log`}</td>
                            </tr>
                            <tr key="today">
                                <td>6</td><td>Logtime today</td><td>{dayLogTime} minutes</td>
                            </tr>
                            <tr key="total">
                                <td>7</td><td>Total logtime</td><td>{totalLogTime} minutes</td>
                            </tr>
                            <tr key="performance">
                                <td>8</td><td>Percentage</td><td>{((dayLogTime/1440)*100).toFixed(2)} %</td>
                            </tr>
                            </tbody>
                        </table>
                    </div>
                </center>
            </div>        
        )
    }
}