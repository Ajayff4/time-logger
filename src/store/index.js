import { createStore, applyMiddleware } from 'redux'
import rootReducer from '../reducers/index'
//import logger from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import rootSaga from '../saga'
import createSagaMiddleware from 'redux-saga'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(rootSaga)

export default store