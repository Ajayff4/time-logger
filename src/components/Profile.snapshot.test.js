import React from 'react'
import renderer from 'react-test-renderer'
import Profile from './Profile'
import { Provider } from 'react-redux';
import store from '../store';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Profile Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Provider store={store}><Router><Profile /></Router></Provider> ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});