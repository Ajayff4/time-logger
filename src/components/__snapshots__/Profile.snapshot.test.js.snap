// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Snapshot Testing... renders component correctly 1`] = `
<div
  id="profileComponent"
>
  <center>
    <div
      className="container"
    >
      <table
        className="table table-dark table-striped"
      >
        <tbody>
          <tr>
            <th>
              #
            </th>
            <th>
              Info
            </th>
            <th>
              Description
            </th>
          </tr>
          <tr>
            <td>
              1
            </td>
            <td>
              Username
            </td>
            <td />
          </tr>
          <tr>
            <td>
              2
            </td>
            <td>
              Fullname
            </td>
            <td />
          </tr>
          <tr>
            <td>
              3
            </td>
            <td>
              Email
            </td>
            <td />
          </tr>
          <tr>
            <td>
              4
            </td>
            <td>
              Completed
            </td>
            <td>
              0 log
            </td>
          </tr>
          <tr>
            <td>
              5
            </td>
            <td>
              Pending
            </td>
            <td>
              0 log
            </td>
          </tr>
          <tr>
            <td>
              6
            </td>
            <td>
              Logtime today
            </td>
            <td>
              0
               minutes
            </td>
          </tr>
          <tr>
            <td>
              7
            </td>
            <td>
              Total logtime
            </td>
            <td>
              0
               minutes
            </td>
          </tr>
          <tr>
            <td>
              8
            </td>
            <td>
              Percentage
            </td>
            <td>
              0.00
               %
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </center>
</div>
`;
